go_modules/spec/dependabot/go_modules/version_spec.rb

```diff
@@ -112,4 +112,20 @@
       end
     end
   end
+
+  describe "<=>" do
+    # These identifiers come from the Go docs: https://go.dev/ref/mod#pseudo-versions
+    it "sorts major versions correctly" do
+      expect(described_class.new("v1.0.0-20231231120000-abcdefabcdef")).to be < described_class.new("v1.0.0")
+    end
+
+    it "sorts pre-release versions correctly" do
+      expect(described_class.new("v1.0.0-pre2.0.20231231120000-abcdefabcdef")).to be >
+                                                                                  described_class.new("v1.0.0-pre2")
+    end
+
+    it "sorts minor versions correctly" do
+      expect(described_class.new("v1.0.1-20231231120000-abcdefabcdef")).to be < described_class.new("v1.0.1")
```

Comment:
```suggestion
      expect(described_class.new("v1.0.1-0.20231231120000-abcdefabcdef")).to be < described_class.new("v1.0.1")
```

Not positive here but from the doc I think we'd expect a `-0` in there?

> vX.Y.(Z+1)-0.yyyymmddhhmmss-abcdefabcdef is used when the base version is a release version like vX.Y.Z. For example, if the base version is v1.2.3, a pseudo-version might be v1.2.4-0.20191109021931-daa7c04131f5.